.github/workflows/java_lint.yml

```
@@ -0,0 +1,29 @@
+name: "Lint Java code"
+
+on:
+  push:
+    paths:
+      - "**.java"
+      - "./.github/workflows/java_lint.yml"
+  pull_request:
+    paths:
+      - "**.java"
+      - "./.github/workflows/java_lint.yml"
```

settings.gradleとgradlewも含めた方がよいかと思います。

また.javaについても、今のままだと/example/java/を追加したとしてもチェックされないので、"voicevox_core_java_api/下の*.java"としておいた方が混乱をきたさなくなると思います。

```suggestion
      - ./crates/voicevox_core_java_api/**/*.java
      - ./crates/voicevox_core_java_api/gradle/**
      - ./crates/voicevox_core_java_api/gradlew
      - ./crates/voicevox_core_java_api/settings.gradle
      - "./.github/workflows/java_lint.yml"
```

---

gradlewとかも含めるようにしました。